LanguageFeatures/Patterns/shared_case_scope_A02_t03.dart

```diff
@@ -56,7 +56,7 @@ String test2(num? x) {
   switch (x) {
     case var a when a is int:
     case var a when a is num:
-      a.expectStaticType<Exactly<num>>();
+      a.expectStaticType<Exactly<num?>>();
```

Comment:
OK, this case will then test that we do _not_ get a promotion when we have promotion chains like `[int, num?]` and `[num, num?]`, because `[int, num?] ^ [num, num?] == [num?]`.